bin/node/cli/tests/telemetry.rs

```diff
@@ -0,0 +1,114 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2020-2021 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use assert_cmd::cargo::cargo_bin;
+use nix::sys::signal::{kill, Signal::SIGINT};
+use nix::unistd::Pid;
+use std::convert::TryInto;
+use std::process;
+use tempfile::tempdir;
+
+pub mod common;
+pub mod websocket_server;
+
+#[async_std::test]
+async fn telemetry_works() {
```

Comment:
Don't you have written a macro for creating async tests in the Substrate context?

Reply:
And in general I don't understand why we use `async-std` here.

Reply:
I will check that tomorrow :sweat_smile:

Reply:
It's a bit complicated with all the `compat` stuff. I couldn't change it easily and it adds more dependency (tokio-util):

```
error[E0599]: no method named `send_response` found for struct `soketto::handshake::Server<'_, tokio::net::TcpStream>` in the current scope
   --> bin/node/cli/tests/websocket_server.rs:133:6
    |
133 |                 .send_response(&{
    |                  ^^^^^^^^^^^^^ method not found in `soketto::handshake::Server<'_, tokio::net::TcpStream>`
    | 
   ::: /home/cecile/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.25/src/net/tcp/stream.rs:58:5
    |
58  |     pub struct TcpStream {
    |     --------------------
    |     |
    |     doesn't satisfy `tokio::net::TcpStream: futures::AsyncRead`
    |     doesn't satisfy `tokio::net::TcpStream: futures::AsyncWrite`
    |
    = note: the method `send_response` exists but the following trait bounds were not satisfied:
            `tokio::net::TcpStream: futures::AsyncRead`
            `tokio::net::TcpStream: futures::AsyncWrite`
```

I think that's good enough for now if that's okay for you.

Reply:
Can you educate me on what the tokio situation is in substrate atm? Is the following correct:
 `libp2p` is executor agnostic (but uses tokio 1.0 in examples/tests), other async subsystems use … tokio 0.2, and `jsonrpc` is stuck on tokio 0.1. And now we add `async-std` here (or is it already used in other tests too)?

If the above is correct we clearly have some tech debt to pay off here. :/

Reply:
We have:
- tokio 0.1 in sc-rpc (and sc-service-test)
- async-std 1 in substrate-prometheus-endpoint, sc-service (dev dependency), sc-network and sc-network-test 
- tokio 0.2 everywhere else

I don't personally mind having multiple different executors in different crates if it is for tests. It's just a dev dependency, it doesn't impact the user. Compiling both isn't a big issue either. :woman_shrugging: